pkg/asset/installconfig/baremetal/host.go

```diff
@@ -0,0 +1,90 @@
+package baremetal
+
+import (
+	"strconv"
+
+	"gopkg.in/AlecAivazis/survey.v1"
+
+	"github.com/openshift/installer/pkg/types/baremetal"
+	"github.com/openshift/installer/pkg/validate"
+)
+
+// Host prompts the user for hardware details about a baremetal host.
+func Host() (*baremetal.Host, error) {
+	var host baremetal.Host
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Name",
+			},
+			Validate: survey.ComposeValidators(survey.Required),
+		},
+	}, &host.Name); err != nil {
+		return nil, err
+	}
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "BMC Address",
+				Help:    "The address for the BMC, e.g. ipmi://192.168.0.1",
+			},
+			Validate: survey.ComposeValidators(survey.Required, uriValidator),
+		},
+	}, &host.BMC.Address); err != nil {
+		return nil, err
+	}
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "BMC Username",
+			},
+			Validate: survey.ComposeValidators(survey.Required),
+		},
+	}, &host.BMC.Username); err != nil {
+		return nil, err
+	}
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Password{
+				Message: "BMC Password",
+			},
+			Validate: survey.ComposeValidators(survey.Required),
+		},
+	}, &host.BMC.Password); err != nil {
+		return nil, err
+	}
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Boot MAC Address",
+			},
+			Validate: survey.ComposeValidators(survey.Required, macValidator),
+		},
+	}, &host.BootMACAddress); err != nil {
+		return nil, err
+	}
+
+	return &host, nil
+}
+
+// isNumber validates if the answer provided is a valid number
```

Comment:
there seems to be no user of this function

Reply:
dang. Sloppy. Updated in the new commit.